src/gdb_main.c

```diff
@@ -846,7 +846,7 @@ void gdb_poll_target(void)
 		gdb_putpacket_f("T%02X", GDB_SIGINT);
 		break;
 	case TARGET_HALT_WATCHPOINT:
-		gdb_putpacket_f("T%02Xwatch:%08X;", GDB_SIGTRAP, watch);
+		gdb_putpacket_f("T%02Xwatch:%08" PRIx32 ";", GDB_SIGTRAP, watch);
```

Comment:
This needs to use `PRIX32` as otherwise some GDBs will fail to properly parse the response.

Reply:
Applied, thanks. Do you know when this notification is triggered (to check/reproduce gdb failing to parse the response)?

Reply:
Whenever you set a watchpoint and it's hit. Something like `awatch *0x<addr>` on an address you know is going to be hit, and then `c` in GDB